File: 0169.majority_element/solution.py
class Solution:

    def majority_element(self, nums):
        total, ans = 0, 0
        for num in nums:
            if total == 0:
                ans = num
            if num == ans:
                total += 1
            else:
                total -= 1
        return ans

    def other_solution(self, nums):
        """ one line"""
        return sorted(nums)[len(nums)//2]

File: 0169.majority_element/solution.java
class solution{
    public int majorityElement(int[] nums){
        int total = 0, ans = 0;
        for (int num: nums){
            if (total == 0)
                ans = num;
            if (ans == num)
                total++;
            else
                total--;
        }
        return ans;
    }

    public int otherSolution(int[] nums){
        Arrays.sort(nums);
        return nums[nums.length/2]
    }
}